pages/blog/event-driven-api-documentation-made-simple-clientside-rendering.md

```diff
@@ -0,0 +1,219 @@
+
+Oh, if you are still not sure about Client-Side Rendering or Server-Side Rendering, then you can read all about it here 👉 [Website rendering for beginners](https://michals-corner.vercel.app/blog/website-rendering-for-beginners) (shameless plug 😉).
+
+I will cover the usage for:
+
+- [React](#react)
+- [Vue](#vue)
+- [Web Components](#wc)
+- [Standalone Bundle](#sb)
+
+All examples will use this same AsyncAPI sample file 👇
+
+```
+{
+  "asyncapi": "2.2.0",
+  "info": {
+    "title": "Account Service",
+    "version": "1.0.0",
+    "description": "This service is in charge of processing user signups"
+  },
+  "channels": {
+    "user/signedup": {
+      "subscribe": {
+        "message": {
+          "$ref": "#/components/messages/UserSignedUp"
+        }
+      }
+    }
+  },
+  "components": {
+    "messages": {
+      "UserSignedUp": {
+        "payload": {
+          "type": "object",
+          "properties": {
+            "displayName": {
+              "type": "string",
+              "description": "Name of the user"
+            },
+            "email": {
+              "type": "string",
+              "format": "email",
+              "description": "Email of the user"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+This is the expected look of the generated document 👇
+
+![document-template](/img/posts/event-driven-API-documentation-made-simple/document-template.webp "Document template")
+
+All usage examples from this article are available to check on [asyncapi-docs-rendering-examples](https://github.com/m1ner79/asyncapi-docs-rendering-examples) repository.
+
+---
+
+### <a name="react"></a>React
+
+If you wish to render documentation from your AsyncAPI file in React application, then you need to use [AsyncAPI React component](https://github.com/asyncapi/asyncapi-react).
+
+1️⃣ To install the **React AsyncAPI component** run the command `npm install --save @asyncapi/react-component@next`.
+
+2️⃣ Now, create **index.js** file and type in:
+
+```
+import React from "react";
+import ReactDOM from "react-dom";
+
+import AsyncApiComponent from "@asyncapi/react-component";
+import "@asyncapi/react-component/styles/default.min.css";
+
+import { specMock } from "./testDoc";
+
+const rootElement = document.getElementById("root");
+ReactDOM.render(<AsyncApiComponent schema={specMock} />, rootElement);
+```
+
+**React AsyncAPI component** is imported on the **line 4**.
+
+If you are happy with AsyncAPI styling then you need to import their CSS pattern with `import "@asyncapi/react-component/styles/default.min.css";`
+
+On the **line 7** is where AsyncAPI sample file is imported.
+
+---
+
+### <a name="vue"></a>Vue
```

Comment:
I made doc for vue probably one year ago and for Vue 2.x (Vue 3.x was still in beta), so please make note in this section that given instruction is for Vue 2.x :)